01_explore_data/nbconverted/explore_cancer_gene_sets.py

```diff
@@ -0,0 +1,388 @@
+from goatools.obo_parser import GODag
+
+obodag = GODag(str(obo_file))
+
+
+# In[10]:
+
+
+from goatools.anno.genetogo_reader import Gene2GoReader
+
+# Read NCBI's gene2go. Store annotations in a list of namedtuples
+objanno = Gene2GoReader(str(gene2go_file), taxids=[9606])
+
+# Get namespace2association where:
+#    namespace is:
+#        BP: biological_process               
+#        MF: molecular_function
+#        CC: cellular_component
+#    assocation is a dict:
+#        key: NCBI GeneID
+#        value: A set of GO IDs associated with that gene
+ns2assoc = objanno.get_ns2assc()
+
+for nspc, id2gos in ns2assoc.items():
+    print("{NS} {N:,} annotated human genes".format(NS=nspc, N=len(id2gos)))
+
+
+# In[11]:
+
+
+# convert NCBI Gene query results for human genes to Python module
+# see: https://github.com/tanghaibao/goatools/blob/main/notebooks/background_genes_ncbi.ipynb
+from goatools.cli.ncbi_gene_results_to_python import ncbi_tsv_to_py
+
+ncbi_tsv = str(cfg.go_data_dir / 'gene_result.txt')
+output_py = 'genes_ncbi_9606_proteincoding.py'
+ncbi_tsv_to_py(ncbi_tsv, output_py)
+
+
+# In[12]:
+
+
+# load human background gene set
+from genes_ncbi_9606_proteincoding import GENEID2NT as GeneID2nt
+
+print(len(GeneID2nt))
+print(list(GeneID2nt.items())[0])
+
+
+# In[13]:
+
+
+# initialize GO enrichment analysis object
+from goatools.goea.go_enrichment_ns import GOEnrichmentStudyNS
+
+goea = GOEnrichmentStudyNS(
+    GeneID2nt.keys(), # list of human protein-coding genes
+    ns2assoc, # gene/GO term associations
+    obodag, # ontology structure
+    propagate_counts=False,
+    alpha=0.05,
+    methods=['fdr_bh']
+)
+
+
+# In[14]:
+
+
+# we need to convert gene symbols to entrez IDs
+from mpmp.utilities.tcga_utilities import get_symbol_map
+
+symbol_to_entrez, old_to_new_entrez = get_symbol_map()
+
+def gene_names_to_ids(gene_names):
+    gene_ids = []
+    for gene in gene_names:
+        try:
+            gene_id = symbol_to_entrez[gene]
+            if gene_id in old_to_new_entrez.keys():
+                gene_id = old_to_new_entrez[gene_id]
+            gene_ids.append(gene_id)
+        except KeyError:
+            print('Gene {} not in ID map'.format(gene), file=sys.stderr)
+            continue
+    return gene_ids
+
+
+# In[15]:
+
+
+# get entrez IDs for Vogelstein genes        
+vogelstein_ids = gene_names_to_ids(vogelstein_df.gene.unique())
+print(len(vogelstein_ids))
+vogelstein_ids[:5]
+
+
+# In[16]:
+
+
+# run GO enrichment analysis for Vogelstein genes
+vogelstein_results = goea.run_study(vogelstein_ids)
+vogelstein_sig_results = [r for r in vogelstein_results if r.p_fdr_bh < 0.05]
+
+
+# In[17]:
+
+
+# save to output directory
+cfg.go_output_dir.mkdir(exist_ok=True)
+
+vogelstein_output_file = cfg.go_output_dir / 'vogelstein_enrichment.tsv'
+goea.wr_tsv(str(vogelstein_output_file), vogelstein_sig_results)
+
+
+# In[18]:
+
+
+# get non-Vogelstein gene set
+all_genes = set.union(*list(label_map.values()))
```

Comment:
Just checking that non-Vogelstein = (Bailey + COSMIC) - Vogelstein?

Reply:
Yep, exactly! It's the combination of the two non-Vogelstein datasets, minus the Vogelstein genes that overlap with those datasets.